lib/level/level-manager.js

```
@@ -0,0 +1,183 @@
+     * will be used as level prototype. Otherwise, path will be matched
+     * against list of patterns registered with setLevelClass methods and
+     * first match will be used. If no mathes found, default level class
+     * will be used.
+     *
+     * @param {String} path path on a filesystem
+     * @param {Object} opts options
+     * @param {Boolean} opts.noCache reload level instance even if its
+     * cached. 
+     */
+    this.createLevel = function(path, opts) {
+        opts = opts || {};
+        if (!opts.noCache && cache[path]) {
+            return cache[path];
+        }
+        var LevelClass,
+            levelJSPath = PATH.join(path, '.bem', 'level.js');
+        if (bemUtil.isRequireable(levelJSPath)) {
+            LevelClass = getLevelClass(levelJSPath, []);
+        } else {
+            LevelClass = findMatchedLevel(path);
+            if (!LevelClass) {
+                LevelClass = Level;
+            }
+        }
+        cache[path] = new LevelClass(path, opts);
+        return cache[path];
+    };
+
+    function getLevelClass(path, stack) {
+        var mixin = getLevelMixin(path, stack);
+        stack.push(path);
+
+        if (isLevelClass(mixin)) {
+            return mixin;
+        }
+
+        if (mixin.Level) {
+            return mixin.Level;
+        }
+        
+        return INHERIT(getLevelBaseClass(mixin, stack), mixin);
+    }
+
+    function getLevelMixin(path, stack) {
+        var mixin;
+        try {
+            mixin = requireLevel(path);
+        } catch(e) {
+            throw new Error(U.format(
+                'level module %s can not be found %s %s%s\n',
+                path,
+                stack.length > 0? 'but required by the level': '',
+                stack.length > 1? 'inheritance tree:\n\t': '',
+                stack.join('\n\t')));
+        }
+
+        if (typeof mixin === 'function') {
+            mixin = mixin(getBem());
+        }
```

probably we should throw if mixin is undefined (e.g. because of empty level.js)